test/exchange_wrapper_test.ts

```diff
@@ -723,33 +716,43 @@ describe('ExchangeWrapper', () => {
                     expect(event).to.not.be.undefined();
                     done();
                 });
-
-                const fillTakerAmountInBaseUnits = new BigNumber(1);
                 await zeroEx.exchange.fillOrderAsync(
                     signedOrder, fillTakerAmountInBaseUnits, shouldCheckTransfer, takerAddress,
                 );
             })();
         });
         it('Should stop watch for events when stopWatchingAsync called on the eventEmitter', (done: DoneCallback) => {
             (async () => {
-                const subscriptionOpts: SubscriptionOpts = {
-                    fromBlock: 0,
-                    toBlock: 'latest',
-                };
                 const eventSubscriptionToBeStopped = await zeroEx.exchange.subscribeAsync(
                     ExchangeEvents.LogFill, subscriptionOpts, indexFilterValues,
                 );
                 eventSubscriptionToBeStopped.watch((err: Error, event: ContractEvent) => {
                     done(new Error('Expected this subscription to have been stopped'));
                 });
                 await eventSubscriptionToBeStopped.stopWatchingAsync();
-                const fillTakerAmountInBaseUnits = new BigNumber(1);
                 await zeroEx.exchange.fillOrderAsync(
                     signedOrder, fillTakerAmountInBaseUnits, shouldCheckTransfer, takerAddress,
                 );
                 done();
             })();
         });
+        it('Should wrap all event args BigNumber instances in a newer version of BigNumber', (done: DoneCallback) => {
+            (async () => {
+                const zeroExEvent = await zeroEx.exchange.subscribeAsync(ExchangeEvents.LogFill, subscriptionOpts,
+                                                                         indexFilterValues);
+                zeroExEvent.watch((err: Error, event: ContractEvent) => {
+                    const args = event.args as LogFillContractEventArgs;
+                    expect(args.filledValueM.isBigNumber).to.be.true();
+                    expect(args.filledValueT.isBigNumber).to.be.true();
+                    expect(args.feeMPaid.isBigNumber).to.be.true();
+                    expect(args.feeTPaid.isBigNumber).to.be.true();
+                    done();
+                });
+                await zeroEx.exchange.fillOrderAsync(
```

Comment:
To really test this out, fill a really big order (an amount that wouldn't fit in a JS `number`). See if this still works for that edge-case.